dartagnan/src/main/java/com/dat3m/dartagnan/encoding/PropertyEncoder.java

```diff
@@ -401,187 +399,9 @@ public TrackableFormula encodeDataRaces() {
     // ======================================================================
     // ======================================================================
 
-    private TrackableFormula encodeDeadlocks() {
-        logger.info("Encoding dead locks");
-        return new LivenessEncoder().encodeLivenessBugs();
+    private TrackableFormula encodeNontermination() {
+        final BooleanFormula hasNontermination = new NonTerminationEncoder(context.getTask(), context).encodeNontermination();
+        return new TrackableFormula(context.getBooleanFormulaManager().not(LIVENESS.getSMTVariable(context)), hasNontermination);
```

Comment:
Should we use `NON_TERMINATION` instead of `LIVENESS`?

Reply:
We specify all properties in the positive (e.g. `program_spec` rather than `spec_violation`), so I think it should be `LIVENESS`. However, the way we check this is by encoding the negative.

Reply:
I was not referring to the positive vs negative part, but rather about the term "liveness" ... at some point we even had some "deadlock/no-deadlock" terms and I thought that "termination" was maybe a more general one.

However, we use liveness for the property and the nice stdout msg when we find a violation, so for consistency we should probably keep it as it is.

Reply:
Hmm, we could use `TERMINATION` as the property. It would be more precise than liveness, since the latter can be understood in many ways.
If I rename it, dependent tools like `Vsyncer` might need updates.

Reply:
Done